solidity/contracts/middleware/InterchainAccountRouter.sol

```diff
@@ -138,6 +127,11 @@ contract InterchainAccountRouter is
             _owner
         );
         require(localDomain == mailbox.localDomain(), "domain mismatch");
+
+        implementation = address(new OwnableMulticall(address(this)));
+        // cannot be stored immutably because it is dynamically sized
```

Comment:
this is actually not true and we have tooling for this though might not be worth the hassle (in this PR or generally)
see https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/solidity/contracts/libs/MetaProxy.sol and usage in https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/fcb1bebb96816126a1f0abe01326edb516efddce/solidity/contracts/libs/StaticMOfNAddressSetFactory.sol#L10